auth/src/main/java/com/vimeo/networking2/requests/interactors/ClientCredentialsInteractor.kt

```diff
@@ -0,0 +1,34 @@
+package com.vimeo.networking2.requests.interactors
+
+import com.vimeo.networking2.enums.GrantType
+import com.vimeo.networking2.enums.ScopeType
+import com.vimeo.networking2.requests.AuthCallback
+import com.vimeo.networking2.requests.AuthService
+import com.vimeo.networking2.requests.VimeoRequest
+import com.vimeo.networking2.utils.enqueueAuthRequest
+
+/**
+ * Interactor which performs auth request with client credentials.
+ *
+ * @param authService   Retrofit API service for auth.
+ * @param authHeaders   Client id and client secret headers.
+ * @param scopes        API scopes to support.
+ */
+internal class ClientCredentialsInteractor(
```

Comment:
this class is so small, and it's also internal - could the `AuthenticatorImpl` just do what this class is doing? There's almost as much setup of the `ClientCredentialsInteractor` required as there is just to talk to the `AuthService` directly.